src/main/java/org/motechproject/mots/management/LocationImporter.java

```diff
@@ -0,0 +1,199 @@
+        continue;
+      }
+
+      District district = new District(cellText);
+
+      if (!districtList.contains(district)) {
+        locationService.createDistrict(new District(cellText));
+      }
+    }
+  }
+
+  private void parseChiefdoms(XSSFSheet sheet) {
+    XSSFRow row;
+    XSSFCell cell;
+    Iterator rows = sheet.rowIterator();
+
+    while (rows.hasNext()) {
+      row = (XSSFRow) rows.next();
+      cell = row.getCell(0);
+      String cellText = cell.getStringCellValue();
+
+      if (cellText.equals(CHIEFDOM_HEADER)) {
+        continue;
+      }
+
+      Chiefdom chiefdom = new Chiefdom(cellText);
+      String parentName = row.getCell(1).getStringCellValue();
+
+      District parent = districtList.stream()
+          .filter(district -> district.getName().equals(parentName))
+          .findFirst().orElseThrow(() -> new RuntimeException(String.format("'%s' Chiefdom parent "
+              + "is not defined properly in spreadsheet", chiefdom.getName())));
+
+      chiefdom.setDistrict(parent);
+
+      if (!chiefdomList.contains(chiefdom)) {
+        locationService.createChiefdom(chiefdom);
+      }
+    }
+  }
+
+  private void parseFacilities(XSSFSheet sheet) {
+    XSSFRow row;
+    XSSFCell cell;
+    Iterator rows = sheet.rowIterator();
+
+    while (rows.hasNext()) {
+      row = (XSSFRow) rows.next();
+      cell = row.getCell(0);
+      String cellText = cell.getStringCellValue();
+
+      if (cellText.equals(FACILITY_HEADER)) {
+        continue;
+      }
+
+      Facility facility = new Facility(cellText);
+      String parentName = row.getCell(1).getStringCellValue();
+
+      Chiefdom parent = chiefdomList.stream()
+          .filter(chiefdom -> chiefdom.getName().equals(parentName))
+          .findFirst().orElseThrow(() -> new RuntimeException(String.format("'%s' Facility parent "
+              + "is not defined properly in spreadsheet", facility.getName())));
+
+      facility.setChiefdom(parent);
+
+      if (!facilityList.contains(facility)) {
+        locationService.createFacility(facility);
+      }
+    }
+  }
+
+  private void parseCommunities(XSSFSheet sheet) {
+    XSSFRow row;
+    XSSFCell cell;
+    Iterator rows = sheet.rowIterator();
+
+    while (rows.hasNext()) {
+      row = (XSSFRow) rows.next();
+      cell = row.getCell(0);
+      String cellText = cell.getStringCellValue();
+
+      if (cellText.equals(COMMUNITY_HEADER)) {
+        continue;
+      }
+
+      Community community = new Community(cellText);
+      String parentName = row.getCell(3).getStringCellValue();
+
+      Facility parent = facilityList.stream()
+          .filter(facility -> facility.getName().equals(parentName))
+          .findFirst().orElseThrow(() -> new RuntimeException(String.format("'%s' Community parent "
+              + "is not defined properly in spreadsheet", community.getName())));
+
+      community.setFacility(parent);
+
+      if (!communityList.contains(community)) {
```

Comment:
I think we should look for this community only in the parent facility list, because community with this name can already exist in different facilities.
The same for Facilities and Chiefdoms.